### PR TITLE
Browser capabilities + schema validation

### PR DESCRIPTION
**Pull request by nirmalashish**

### CODE REVIEW

File: src/main/java/com/znsio/e2e/runner/Setup.java [hunk 1]
```diff
@@ -90,19 +88,25 @@
     private static final String LAUNCH_NAME_SUFFIX = "LAUNCH_NAME_SUFFIX";
     private static final String APPIUM_UI_AUTOMATOR2_SERVER = "io.appium.uiautomator2.server";
     private static final String APPIUM_SETTINGS = "io.appium.settings";
-    private static final Logger LOGGER = Logger.getLogger(Setup.class.getName());
     public static final String WEBDRIVER_MANAGER_PROXY_URL = "WEBDRIVER_MANAGER_PROXY_URL";
+    private static final String REMOTE_WEBDRIVER_GRID_PORT_KEY = "REMOTE_WEBDRIVER_GRID_PORT_KEY";
+    public static final String BROWSER_CONFIG_FILE = "BROWSER_CONFIG_FILE";
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/runner/Setup.java:93]: Do we need these to be public?

File: src/main/java/com/znsio/e2e/runner/Setup.java [hunk 2]
```diff
@@ -90,19 +88,25 @@
     private static final String LAUNCH_NAME_SUFFIX = "LAUNCH_NAME_SUFFIX";
     private static final String APPIUM_UI_AUTOMATOR2_SERVER = "io.appium.uiautomator2.server";
     private static final String APPIUM_SETTINGS = "io.appium.settings";
-    private static final Logger LOGGER = Logger.getLogger(Setup.class.getName());
     public static final String WEBDRIVER_MANAGER_PROXY_URL = "WEBDRIVER_MANAGER_PROXY_URL";
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/runner/Setup.java:91]: Do we need these to be public?

File: src/main/java/com/znsio/e2e/runner/Setup.java [hunk 3]
```diff
@@ -90,19 +88,25 @@
...
+    private static final String REMOTE_WEBDRIVER_GRID_PORT_KEY = "REMOTE_WEBDRIVER_GRID_PORT_KEY";
+    public static final String BROWSER_CONFIG_FILE = "BROWSER_CONFIG_FILE";
+    public static final String BROWSER_CONFIG_FILE_CONTENTS = "BROWSER_CONFIG_FILE_CONTENTS";
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/runner/Setup.java:94]: Do we need these to be public?

File: src/main/java/com/znsio/e2e/runner/Setup.java [hunk 4]
```diff
@@ -90,19 +88,25 @@
...
+
     static Map<String, Map> environmentConfiguration;
     static Map<String, Map> testDataForEnvironment;
     static Map applitoolsConfiguration = new HashMap<>();
-    private static final String REMOTE_WEBDRIVER_GRID_PORT_KEY = "REMOTE_WEBDRIVER_GRID_PORT_KEY";
+
     private final Properties properties;
-    private final String DEFAULT_LOG_PROPERTIES_FILE = "./src/main/resources/defaultLog4j.properties";
+    private final String DEFAULT_LOG_PROPERTIES_FILE = "/defaultLog4j.properties";
+    public static final String DEFAULT_BROWSER_CONFIG_FILE = "/default_browser_config.json";
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/runner/Setup.java:103]: Do we need these to be public?

File: src/main/java/com/znsio/e2e/runner/Setup.java (hunk 1 above)
Comment (nirmalashish) [reply src/main/java/com/znsio/e2e/runner/Setup.java]: pushed a change and made them package-protected

File: src/main/java/com/znsio/e2e/runner/Setup.java (hunk 3 above)
Comment (nirmalashish) [reply src/main/java/com/znsio/e2e/runner/Setup.java]: pushed a change and made them package-protected

**Pull request merged**